src/js/actions/layers.js

```diff
@@ -1377,45 +1382,154 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard \
+     * 
+     * @private 
+     * @param {Immutable.List.<Layer>} artboards
+     * @return {Layer}
+     */
+    var _findRightMostArtboard = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+            if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                return currentLayer;
+            } else {
+                return selectedLayer;
+            }
+        }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     *
+     * @private
+     * @param {object} specs 
+     * @param {Immutable.List.<Layer>} artboards
+     * @param {Layer} layerRef
```

Comment:
This parameter, and the fact that it's also in the return object, is weird. My intuition says that this is not the right way to do this. I see that in one path through this function you're changing the ref, but maybe you can just return a flag that would be used by the caller to do something with the ref instead?